### ISSUE TITLE
Uninformative summary of types in Julia 0.7

### ISSUE THREAD

**Issue by giordano** [JuliaLang/julia#22156]
Maybe it has been already reported before, but I couldn't find a ticket for the following issue.  The help for certain types isn't informative at all in Julia 0.7 (and probably Julia 0.6 as well):
```julia
help?> Range
search: Range range RangeIndex nzrange UnitRange StepRange StepRangeLen ClusterManager trailing_zeros trailing_ones OrdinalRange

  No documentation found.

  Base.Range is of type UnionAll.

  Summary:

  struct UnionAll <: Type{T}

  Fields:

  var  :: TypeVar
  body :: Any

  Supertype Hierarchy:

  UnionAll <: Type{T} <: Any

help?> UnitRange
  No documentation found.

  Base.UnitRange is of type UnionAll.

  Summary:

  struct UnionAll <: Type{T}

  Fields:

  var  :: TypeVar
  body :: Any

  Supertype Hierarchy:

  UnionAll <: Type{T} <: Any
```

Note that this is a regression compared to Julia 0.5, where instead I get:
```julia
help?> Range
search: Range range RangeIndex nzrange UnitRange StepRange histrange FloatRange ClusterManager trailing_zeros trailingsize trailing_ones

  No documentation found.

  Summary:

  abstract Range{T} <: AbstractArray{T,1}

  Subtypes:

  FloatRange{T<:AbstractFloat}
  LinSpace{T<:AbstractFloat}
  OrdinalRange{T,S}

help?> UnitRange
search: UnitRange AbstractUnitRange

  No documentation found.

  Summary:

  immutable UnitRange{T<:Real} <: AbstractUnitRange{T<:Real}

  Fields:

  start :: T<:Real
  stop  :: T<:Real
```

**Comment by yuyichao** [comment]
Dup of https://github.com/JuliaLang/julia/issues/21029 (see second comment.)

**Comment by giordano** [comment]
Oh, thanks.  I skimmed through the bugs labeled with `doc` reading the titles, that's why I couldn't find the issue.  Anyway, note that it's a regression, so it should be tagged as such (or perhaps documented in the NEWS if won't be addressed before 0.6 release).
